includes/class-wc-payments.php

```diff
@@ -510,6 +512,25 @@ public static function register_checkout_gateway( $payment_method_registry ) {
 		$payment_method_registry->register( new WC_Payments_Blocks_Payment_Method() );
 	}
 
+	/**
+	 * Handles upgrade routines.
+	 */
+	public static function install() {
+		self::add_woo_admin_notes();
+		if ( WCPAY_VERSION_NUMBER !== get_option( 'woocommerce_woocommerce_payments_version' ) ) {
+			do_action( 'woocommerce_woocommerce_payments_updated' );
+			self::update_plugin_version();
+		}
+	}
+
+	/**
+	 * Updates the plugin version in db
+	 */
+	public static function update_plugin_version() {
+		delete_option( 'woocommerce_woocommerce_payments_version' );
+		update_option( 'woocommerce_woocommerce_payments_version', WCPAY_VERSION_NUMBER );
```

Comment:
I don't think it's necessary to delete the previous value before calling `update_option`. The [docs for `update_option`](https://developer.wordpress.org/reference/functions/update_option/) say:

> Updates the value of an option that was already added.
> [...]
> If the option does not exist, it will be created.

Reply:
I remember being in doubt about this one too 🤔 but I just copied [what was implemented in Stripe](https://github.com/woocommerce/woocommerce-gateway-stripe/blob/7a25f88e6108560174b3cb7b3759fcff726fc868/woocommerce-gateway-stripe.php#L194). I don't really know why they used that delete_option... I think it makes sense to remove it.